src/bin/vip-logs.js

```diff
@@ -43,14 +42,89 @@ export async function getLogs( arg: string[], opt ): Promise<void> {
 
 	await trackEvent( 'logs_command_success', {
 		...trackingParams,
-		logs_output: logs.length,
+		total: logs.nodes.length,
 	} );
 
-	if ( ! logs.length ) {
+	if ( ! logs.nodes.length ) {
 		console.error( 'No logs found' );
 		return;
 	}
 
+	printLogs( logs.nodes, opt.format );
+}
+
+export async function followLogs( opt ): Promise<void> {
+	let after = null;
+	let isFirstRequest = true;
+	// How many times have we polled?
+	let requestNumber = 0;
+
+	const trackingParams = getBaseTrackingParams( opt );
+
+	// Set an initial default delay
+	let delay = DEFAULT_POLLING_DELAY_IN_SECONDS;
+
+	while ( true ) {
+		const limit = isFirstRequest ? opt.limit : LIMIT_MAX;
+
+		requestNumber++;
+		trackingParams.request_number = requestNumber;
+		trackingParams.request_delay = delay;
+		trackingParams.limit = limit;
+
+		let logs;
+		try {
+			logs = await logsLib.getRecentLogs( opt.app.id, opt.env.id, opt.type, limit, after );
+
+			await trackEvent( 'logs_command_follow_success', {
+				...trackingParams,
+				total: logs?.nodes.length,
+			} );
+		} catch ( error ) {
+			await trackEvent( 'logs_command_follow_error', { ...trackingParams, error: error.message } );
+
+			// If the first request fails we don't want to retry (it's probably not recoverable)
+			if ( isFirstRequest ) {
+				console.error( 'Error fetching initial logs' );
+				break;
+			}
+			// Increase the delay on errors to avoid overloading the server, up to a max of 5 minutes
+			delay += DEFAULT_POLLING_DELAY_IN_SECONDS;
+			delay = Math.min( delay, MAX_POLLING_DELAY_IN_SECONDS );
+			console.error( `Failed to fetch logs. Trying again in ${ delay } seconds` );
```

Comment:
[2] Shouldn't we print the error message? If the user is hitting a rate limit for example, they will never know.
```suggestion
			console.error( `Failed to fetch logs: ${ error.message }` );
			console.error( `Trying again in ${ delay } seconds` );
```

Reply:
Skipping this as per the explanation in the other comment